homeworks/src/homework12/Apple.java

```diff
@@ -0,0 +1,8 @@
+package homework12;
+
+public class Apple extends Fruit{
+
+    public Apple(float weight) {
```

Comment:
you can add the waight as a constant to avoid putting in every time in the constructor
`public static float APPLE_WEIGHT = 1F;`
the same for Orange

Reply:
Weight of the every piece can vary, thus I'd prefer implicitly point it out for every Fruit subclass in constructor.

Reply:
in case you are looking class level solution, you can try something like this
```
public class Apple extends Fruit {
  public static final float APPLE_WEIGHT = 1F;

  public Apple() {
    super(APPLE_WEIGHT);
  }
}
```
by doing so, you ensure that the wrong value is not passed to the constructor

Reply:
sounds reasonable! fixed! Thank You!